src/main/java/seedu/address/model/person/Person.java

```diff
@@ -25,29 +26,33 @@ public class Person {
     // Data fields
     private final Optional<Address> address;
     private final Set<Tag> tags = new HashSet<>();
+    private final ModuleRoleMap moduleRoleMap;
 
     /**
      * Every field must be present and not null.
      */
-    public Person(Name name, Phone phone, Email email, Address address, Set<Tag> tags) {
-        requireAllNonNull(name, phone, email, address, tags);
+    public Person(Name name, Phone phone, Email email, Optional<Address> address, Set<Tag> tags) {
```

Comment:
currently our add command plans to have at least one `r/modulecode-role` parameter

in the case that we want to allow it to be an optional parameter in the future, we can construct a `ModuleRoleMap` then call the constructor

so maybe we might not need this constructor?